.github/workflows/generate-release-script.yml

```diff
@@ -0,0 +1,35 @@
+name: Generate Release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  generate_release:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+
+    - name: Set up environment
+      run: echo "TIMESTAMP=$(date -d '${{ github.event.head_commit.timestamp }}' +%Y)" >> $GITHUB_ENV
+
+    - name: Zip the repository
+      run: zip -r Wall-E.zip .
+
+    - name: Upload Release Asset
+      uses: softprops/action-gh-release@v0.1.15
+      with:
+        files: Wall-E.zip
+        tag_name: ${{ env.TIMESTAMP }}
```

Comment:
Are you sure that we want to use the timestamp for the release?

Reply:
The timestamp variable creates a tag with the year of trigger of the action. Check above for it’s logic.